### PR TITLE
Introduced new featurizer metadata

### PR DESCRIPTION
**Pull request by davidbrownellWork**
- release_version (required)
- updated_version
- deprecated_version
- is_stateless_transformer

### CODE REVIEW

File: src/Tools/CodeGenerator/Featurizers.SimpleSchema [hunk 1]
```diff
@@ -24,10 +24,21 @@
 <featurizers +>:
     [name string]
     [estimator_name string]
+
+    [release_version string]
+    [updated_version string ?]
+    [deprecated_version string ?]
```
Comment (michaelgsharp) [inline src/Tools/CodeGenerator/Featurizers.SimpleSchema]: Whats the difference between release_version and updated_version?

File: src/Tools/CodeGenerator/Featurizers.yaml [hunk 2]
```diff
@@ -2,6 +2,10 @@ featurizers:
   # ----------------------------------------------------------------------
   - name: CatImputerFeaturizer
     estimator_name: CatImputerEstimator
+
+    release_version: 0.3.0
+    deprecated_version: 0.3.6
```
Comment (michaelgsharp) [inline src/Tools/CodeGenerator/Featurizers.yaml]: Are these flipped? If I am reading right it seems the deprecated_version is newer then the release_version.

File: src/Tools/CodeGenerator/Featurizers.SimpleSchema (hunk 1 above)
Comment (davidbrownellWork) [reply src/Tools/CodeGenerator/Featurizers.SimpleSchema]: Release_version is when the featurizer was released, updated_version is the version when the featurizer was last updated after its release.

File: src/Tools/CodeGenerator/Featurizers.yaml (hunk 2 above)
Comment (davidbrownellWork) [reply src/Tools/CodeGenerator/Featurizers.yaml]: That's correct. deprecated_version is an indication that the featurizer should no longer be used (which implies that it has already been released). Let me know if you can think of a less confusing name.

**Pull request merged**